tide/core/PowerTimer.h

```diff
@@ -0,0 +1,74 @@
+#ifndef POWERTIMER_H
+#define POWERTIMER_H
+
+#include "serialization/includes.h"
+#include "types.h"
+
+#include <boost/enable_shared_from_this.hpp>
+
+#include <QObject>
+#include <QTimer>
+
+/**
+ * Inform user about inactivity timeout and allow to stop the power off process.
+ */
+class PowerTimer : public QObject,
+                   public boost::enable_shared_from_this<PowerTimer>
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(PowerTimer)
+
+public:
+    /**
+     * Construct a timer used to power off Planar equipment.
+     * @param timeout value of the timer in ms.
+     */
+    PowerTimer(const int timeout);
+
+    /** Default constructor */
```

Comment:
Document that this creates an "invalid", read-only timer to be used only on Wall processes